pp/go/relabeler/head/head.go

```diff
@@ -663,9 +676,18 @@ func (h *Head) CopySeriesFrom(other relabeler.Head) {
 // Close wals and clear metrics.
 func (h *Head) Close() error {
 	h.memoryInUse.DeletePartialMatch(prometheus.Labels{"generation": strconv.FormatUint(h.generation, 10)})
+
 	var err error
 	for _, s := range h.shards {
 		err = errors.Join(err, s.wal.Close())
+
+		if s.unloadedDataStorage != nil {
+			err = errors.Join(s.unloadedDataStorage.Close())
```

Comment:
here you replace error instead of append new error to it
```suggestion
			err = errors.Join(err, s.unloadedDataStorage.Close())
```